src/compiler/types/validate-primary-package-output-target.ts

```diff
@@ -0,0 +1,222 @@
+    const eligiblePrimaryTargets: d.PrimaryPackageOutputTarget[] = [];
+    const nonPrimaryTargets: d.OutputTarget[] = [];
+
+    // Push each output target in the config into its respective
+    // classification for validation messages
+    // Using a `foreach` prevents us from iterating over
+    // the array multiple times
+    config.outputTargets.forEach((ref) => {
+      if (isPrimaryPackageOutputTarget(ref)) {
+        eligiblePrimaryTargets.push(ref);
+      } else {
+        nonPrimaryTargets.push(ref);
+      }
+    });
+
+    // If there are no output targets designated as "primary", then we should warn the user
+    // to designate one. In this case, we aren't gonna do any validation
+    if (eligiblePrimaryTargets.length) {
+      const targetsMarkedToValidate = eligiblePrimaryTargets.filter((ref) => ref.isPrimaryPackageOutputTarget);
+
+      if (targetsMarkedToValidate.length) {
+        // A user should only designate one target to validate against
+        // Log a warning if they try to have more than one, but we'll only validate
+        // the first one in the array
+        if (targetsMarkedToValidate.length > 1) {
+          logValidationWarning(
+            buildCtx,
+            `Your Stencil config has multiple output targets with 'isPrimaryPackageOutputTarget: true'. Stencil does not support validating 'package.json' fields for multiple output targets. Please updated your Stencil config to only assign one primary package output target. For now, Stencil will use the first primary target it finds.`
+          );
+        }
+
+        // Validate shared fields
+        // Currently, this is only `module` and `types`
+        // We only validate the first target that is designated
+        const targetToValidate = targetsMarkedToValidate[0];
+        const recommendedConfig = PRIMARY_PACKAGE_TARGET_CONFIGS[targetToValidate.type];
+        if (recommendedConfig != null) {
+          validateModulePath(config, compilerCtx, buildCtx, recommendedConfig, targetToValidate);
+          validateTypesPath(config, compilerCtx, buildCtx, recommendedConfig, targetToValidate);
+        }
+      } else {
+        logValidationWarning(
+          buildCtx,
+          `Your Stencil project has not assigned a primary package output target. Stencil recommends that you assign a primary output target so it can validate values for fields in your project's 'package.json'`
+        );
+      }
+    }
+
+    // Log a warning if any targets that cannot be validated were marked as "primary"
+    if (nonPrimaryTargets.length && nonPrimaryTargets.some((ref: any) => ref.isPrimaryPackageOutputTarget)) {
+      logValidationWarning(
+        buildCtx,
+        `Your Stencil project has assigned one or more un-validated output targets as the primary package output target. No validation will take place. Please remove the 'isPrimaryPackageOutputTarget' flag from the following output targets in your Stencil config: ${nonPrimaryTargets
```

Comment:
In my local testing, I made the following changes to my stencil config and ran build:

![Screenshot 2023-05-24 at 1 06 58 PM](https://github.com/ionic-team/stencil/assets/1930213/99722137-fef5-4fae-b5ff-8700bfec2e5b)

Should we suppress the first message if we print the second? The two seem to be conflicting with one another

Reply:
I don't think there's necessarily anything wrong with logging both messages since there are technically two configuration errors. This is something I think we roll with for now (with the added link to docs in each message) and react if we hear feedback that the messages are confusing, conflicting, or whatever